src/main/java/io/openliberty/tools/common/plugins/util/ServerFeatureUtil.java

```diff
@@ -102,21 +103,59 @@
     /**
      * Get the set of features defined in the server.xml
      * @param serverDirectory The server directory containing the server.xml
+     * @param serverXmlFile The server.xml file
      * @param libertyDirPropFiles Map of Liberty directory properties to the actual File for each directory
+     * @param dropinsFilesToIgnore A set of file names under configDropins/overrides or configDropins/defaults to ignore
      * @return the set of features that should be installed from server.xml, or empty set if nothing should be installed
+     *         or null if there are no valid xml files or they have no featureManager section
      */
-    public Set<String> getServerFeatures(File serverDirectory, Map<String,File> libertyDirPropFiles) {
+     public Set<String> getServerFeatures(File serverDirectory, File serverXmlFile, Map<String,File> libertyDirPropFiles, Set<String> dropinsFilesToIgnore) {
         if (libertyDirPropFiles != null) {
             libertyDirectoryPropertyToFile = new HashMap(libertyDirPropFiles);
         } else {
             warn("The properties for directories are null and could lead to server include files not being processed for server features.");
             libertyDirectoryPropertyToFile = new HashMap<String,File>();
         }
         Properties bootstrapProperties = getBootstrapProperties(new File(serverDirectory, "bootstrap.properties"));
-        Set<String> result = getConfigDropinsFeatures(null, serverDirectory, bootstrapProperties, "defaults");
-        result = getServerXmlFeatures(result, new File(serverDirectory, "server.xml"), bootstrapProperties, null);
+        Set<String> result = getConfigDropinsFeatures(null, serverDirectory, bootstrapProperties, "defaults", dropinsFilesToIgnore);
+        if (serverXmlFile == null) {
+            serverXmlFile = new File(serverDirectory, "server.xml");
+        }
+        result = getServerXmlFeatures(result, serverXmlFile, bootstrapProperties, null);
```

Comment:
One potential problem to passing in the serverXmlFile is the case where it has `include` elements with relative paths to other config files that are located in their configDir. When the serverXmlFile gets copied to the target server along with the contents of the configDir, everything would resolve correctly. But potentially they specify a serverXmlFile from a different location, and now the includes cannot be found correctly to find all the features. This is related to the calls to `parseIncludeNode` in this class.